include/envoy/stream_info/filter_state.h

```diff
@@ -49,8 +56,21 @@ class FilterState {
    * single authoritative source for each piece of immutable data stored in
    * FilterState.
    */
-  virtual void setData(absl::string_view data_name, std::unique_ptr<Object>&& data,
-                       StateType state_type) PURE;
+  virtual void setData(absl::string_view data_name, std::shared_ptr<Object> data,
+                       StateType state_type, LifeSpan life_span = LifeSpan::FilterChain) PURE;
```

Comment:
nit: let's not use a default param here and make the caller clearly specify what they expect.

Reply:
Done.

Reply:
Hey Matt, WDYT of having a default param and a TODO to get rid of it?  This is the sort of API change IMO it's nice to land non-required and email about, give people a few weeks to change their code, and then make required?  Unless it's just us that appreciates the option of delayed clean-up :-P
@yavlasov because either way this goes, you should coordinate with @penguingao on how he can do pre-work for import - I think this will be a messy one.

Reply:
Sure that's fine as long as we track cleanup.

Reply:
OK. I'll revert, leave a TODO for myself and open an issue to tracks the cleanup. SG?